### PR TITLE
TP2022T16-53 feature flags

### PR DESCRIPTION
**Pull request by rastislav-balcercik**

### CODE REVIEW

**Comment by Sibyx** [comment]
Hi @rastiQ, 
I am appreciating your minimalistic approach and idea of saving database storage. Introducing new multiple columns for storage flags is also a bad idea (as you mentioned because of the model flexibility/readability). I would like to point out that if you use a varchar and string masks for such data, it will be hard to ask questions such as: 

- which documents are allowed to be printed?
- how many documents use single-page retrieval?

If you want to still use only one column check [hstore](https://www.postgresql.org/docs/current/hstore.html) or [jsonb](https://www.postgresql.org/docs/current/functions-json.html) database types. These non-relational types are indexable so there is no performance issue and you can perform queries on them. Performing such masking queries on varchar will be expensive.

**Comment by rastislav-balcercik** [comment]
> If you want to still use only one column check [hstore](https://www.postgresql.org/docs/current/hstore.html) or [jsonb](https://www.postgresql.org/docs/current/functions-json.html) database types. These non-relational types are indexable so there is no performance issue and you can perform queries on them. Performing such masking queries on varchar will be expensive.

@Sibyx Sure, but is this approach using non-relational acceptable? Or should I go through this problem again and find different solution without saving database storage?

**Comment by Sibyx** [comment]
> @Sibyx Sure, but is this approach using non-relational acceptable? Or should I go through this problem again and find different solution without saving database storage?

@rastiQ, yep this approach is often in such use-cases. Data are not stored as a plaintext so you don't have to worry about performance and storage issues.

**Comment by rastislav-balcercik** [comment]
> Data are not stored as a plaintext so you don't have to worry about performance and storage issues.

Check my latest commit in this PR, let me know if It makes sense.

File: docs/specification/feature_flags.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  zobrazenie tlačidla pre tlač dokumentu
+
+- **share** - povoľuje na front-ende zdieľanie alebo export dokumentu
+
+- **citations** - povoľuje čítateľovi generovať citácie
+
+- **render_type** - nastavením tohto povolenia na **1** spôsobí, že dokument
+  sa na front-ende vyrenderuje po stranách (používateľ uvidí len jednu stranu). Ak je nastavené na
+  **2** - dokument sa vyrenderuje celý a používateľ môže scrollovať celý dokument naraz. Číslovanie je
+  zameniteľné za iný typ, napr. `string`. V budćnosti je možné pridávať ďalšie spôsoby renderovania.
```
Comment (Sibyx) [inline docs/specification/feature_flags.md:40]: Feel free to use `string` enumeration values

**Review by Sibyx — APPROVED** [review]

**Pull request merged**